src/configuration-requestor.spec.ts

```diff
@@ -111,97 +115,309 @@ describe('ConfigurationRequestor', () => {
+        expect(nikeGenderIdentityCoefficient.valueCoefficients).toStrictEqual({
+          female: 0.5,
+          male: -0.5,
+        });
+
+        // Just spot check the adidas parameters
+        expect(bannerCoefficients['adidas'].subjectNumericCoefficients).toHaveLength(0);
+        expect(
+          bannerCoefficients['adidas'].subjectCategoricalCoefficients[0].valueCoefficients[
+            'female'
+          ],
+        ).toBe(0);
+
+        const coldStartBandit = banditModelStore.get('cold_start_bandit');
+        expect(coldStartBandit?.banditKey).toBe('cold_start_bandit');
+        expect(coldStartBandit?.modelName).toBe('falcon');
+        expect(coldStartBandit?.modelVersion).toBe('cold start');
+        const coldStartModelData = coldStartBandit?.modelData;
+        expect(coldStartModelData?.gamma).toBe(1);
+        expect(coldStartModelData?.defaultActionScore).toBe(0);
+        expect(coldStartModelData?.actionProbabilityFloor).toBe(0);
+        expect(coldStartModelData?.coefficients).toStrictEqual({});
       });
 
-      // Just spot check the adidas parameters
-      expect(bannerCoefficients['adidas'].subjectNumericCoefficients).toHaveLength(0);
-      expect(
-        bannerCoefficients['adidas'].subjectCategoricalCoefficients[0].valueCoefficients['female'],
-      ).toBe(0);
-
-      const coldStartBandit = banditModelStore.get('cold_start_bandit');
-      expect(coldStartBandit?.banditKey).toBe('cold_start_bandit');
-      expect(coldStartBandit?.modelName).toBe('falcon');
-      expect(coldStartBandit?.modelVersion).toBe('cold start');
-      const coldStartModelData = coldStartBandit?.modelData;
-      expect(coldStartModelData?.gamma).toBe(1);
-      expect(coldStartModelData?.defaultActionScore).toBe(0);
-      expect(coldStartModelData?.actionProbabilityFloor).toBe(0);
-      expect(coldStartModelData?.coefficients).toStrictEqual({});
-    });
+      it('Will not fetch bandit parameters if there is no store', async () => {
+        configurationRequestor = new ConfigurationRequestor(httpClient, flagStore, null, null);
+        await configurationRequestor.fetchAndStoreConfigurations();
+        expect(fetchSpy).toHaveBeenCalledTimes(1);
+      });
 
-    it('Will not fetch bandit parameters if there is no store', async () => {
-      configurationRequestor = new ConfigurationRequestor(httpClient, flagStore, null, null);
-      await configurationRequestor.fetchAndStoreConfigurations();
-      expect(fetchSpy).toHaveBeenCalledTimes(1);
+      it('Should not fetch bandits if model version is un-changed', async () => {
+        await configurationRequestor.fetchAndStoreConfigurations();
+        expect(fetchSpy).toHaveBeenCalledTimes(2); // Once for UFC, another for bandits
+
+        await configurationRequestor.fetchAndStoreConfigurations();
+        expect(fetchSpy).toHaveBeenCalledTimes(3); // Once just for UFC, bandits should be skipped
+      });
+
+      const warmStartBanditReference = {
```

Comment:
these will be hoised to the top-level describe. Consider putting them in either in it's new nested `describe()` block along with the three tests that use it.